### ISSUE TITLE
A loss function that can directly optimise for AUPRC scores.

### ISSUE THREAD

**Issue by arvind0422** [arvind0422/Chromputer_Arvind#4]
I'm trying to incorporate this https://github.com/tensorflow/models/tree/master/research/global_objectives from the paper by Google engineers: https://arxiv.org/pdf/1608.04802.pdf
